core/primitives/src/runtime/parameter_table.rs

```diff
@@ -0,0 +1,525 @@
+            key = key.strip_prefix(remove_prefix).unwrap_or(key);
+            if let Some(value) = self.get(*param) {
+                json.insert(key.to_owned(), value.clone());
+            }
+        }
+        json.into()
+    }
+
+    fn get(&self, key: Parameter) -> Option<&serde_json::Value> {
+        self.params.get(&key)
+    }
+
+    fn fee_json(&self, key: FeeParameter) -> serde_json::Value {
+        json!( {
+            "send_sir": self.get(format!("{key}_send_sir").parse().unwrap()),
+            "send_not_sir": self.get(format!("{key}_send_not_sir").parse().unwrap()),
+            "execution": self.get(format!("{key}_execution").parse().unwrap()),
+        })
+    }
+}
+
+impl ParameterTableDiff {
+    pub(crate) fn from_txt(arg: &str) -> Result<ParameterTableDiff, InvalidConfigError> {
+        let parameters = txt_to_key_values(arg)
+            .map(|result| {
+                let (typed_key, value) = result?;
+                if let Some((before, after)) = value.split_once("->") {
+                    Ok((
+                        typed_key,
+                        (
+                            parse_parameter_txt_value(before.trim())?,
+                            parse_parameter_txt_value(after.trim())?,
+                        ),
+                    ))
+                } else {
+                    Ok((
+                        typed_key,
+                        (serde_json::Value::Null, parse_parameter_txt_value(value.trim())?),
+                    ))
+                }
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok(ParameterTableDiff { params: BTreeMap::from_iter(parameters) })
+    }
+}
+
+fn txt_to_key_values(
+    arg: &str,
+) -> impl Iterator<Item = Result<(Parameter, &str), InvalidConfigError>> {
+    arg.lines()
+        .enumerate()
+        .filter_map(|(nr, line)| {
+            // ignore comments and empty lines
+            let trimmed = line.trim();
+            if trimmed.starts_with("#") || trimmed.is_empty() {
+                None
+            } else {
+                Some((nr, trimmed))
+            }
+        })
+        .map(|(nr, trimmed)| {
+            let (key, value) = trimmed
+                .split_once(":")
+                .ok_or(InvalidConfigError::NoSeparator(nr + 1, trimmed.to_owned()))?;
+            let typed_key: Parameter = key
+                .trim()
+                .parse()
+                .map_err(|err| InvalidConfigError::UnknownParameter(err, key.to_owned()))?;
+            Ok((typed_key, value))
+        })
+}
+
+/// Parses a value from the custom format for runtime parameter definitions.
+///
+/// A value can be a positive integer or a string, both written without quotes.
+/// Integers can use underlines as separators (for readability).
+fn parse_parameter_txt_value(value: &str) -> Result<serde_json::Value, InvalidConfigError> {
+    if value.is_empty() {
+        return Ok(serde_json::Value::Null);
+    }
```

Comment:
this check feels like somethig worth pushing to the caller perhaps. Just a gut automatic reaction, maybe this is wrong in context :)

Reply:
Not sure about this. Most of the parameters are number types and I would rather have it as `null` when they are unspecified. Also, serde will not like that, I think.